### DISCUSSION TITLE
Unable to switch chain in Metamask with WalletConnect v2

### DISCUSSION THREAD

**Discussion by 0xvidejk** [rainbow-me/rainbowkit#1335] [Q&A] [answered]
Hello! We have a dapp, where user can switch between chains. If user using mobile device connects to our dapp with metamask, wallet is connected, but if tries to switch the chain, nothing happens in metamask (popup with switch button doesn't appear)
In addition, it is impossible to connect metamask through: ` metaMaskWallet({
        projectId,
        chains,
      }),` only using ` walletConnectWallet({
        projectId,
        chains})`
        
what am I doing wrong?

**Comment by DanielSinclair** [comment, marked as answer]
@0xvidejk Switching networks appears to be working fine in MetaMask. Let me know if you're still experiencing an issue there. The other issue with connecting to MetaMask Mobile via WalletConnect is now fixed in [0.12.17](https://github.com/rainbow-me/rainbowkit/releases/tag/%40rainbow-me%2Frainbowkit%400.12.17) and [1.0.5](https://github.com/rainbow-me/rainbowkit/releases/tag/%40rainbow-me%2Frainbowkit%401.0.5).

**Reply by 0xvidejk** [reply]
seems like an issue with switching chains still exists. waiting for Metamask updates for v7.4.0
